commit-graph.c

```diff
@@ -381,6 +385,20 @@ struct commit_graph *parse_commit_graph(void *graph_map, size_t graph_size)
 				graph->chunk_commit_data = data + chunk_offset;
 			break;
 
+		case GRAPH_CHUNKID_GENERATION_DATA:
```

Comment:
(Commenting on the commit message here)

> We test the overflow-related code with the following repo history:
> 
>            F - N - U
>           /         \
> U - N - U            N
>          \          /
> 	  N - F - N
> 
> Where U is a commit with committer date of zero seconds since Unix
> epoch, N is a commit with committer date of 1112354055 (default
> committer date for the test suite) seconds since Unix epoch and F is
> a commit with committer date of (2 ^ 31 - 2) seconds since Unix epoch.

My nitpick here is that you use `F`, `N`, and `U` to indicate a property of the commits, but they are not "a commit". Instead, I would reword this as "The commits denoted by 'U' have committer date of zero seconds...".